vcf-taggable-text-area/src/main/java/org/vaadin/addons/componentfactory/tta/BaseFilterListSelector.java

```diff
@@ -0,0 +1,77 @@
+/*-
+ * #%L
+ * Taggable Text Area
+ * %%
+ * Copyright (C) 2024 Vaadin Ltd
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package org.vaadin.addons.componentfactory.tta;
+
+import com.vaadin.flow.component.HasStyle;
+import com.vaadin.flow.component.Key;
+import com.vaadin.flow.component.customfield.CustomField;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.data.selection.SingleSelect;
+import com.vaadin.flow.data.value.ValueChangeMode;
+import com.vaadin.flow.function.SerializableBiFunction;
+import java.util.List;
+
+@SuppressWarnings("serial")
+public abstract class BaseFilterListSelector<T> extends CustomField<T>
+    implements SingleSelect<CustomField<T>, T>, HasStyle {
+
+  protected TextField filter = new TextField();
```

Comment:
Is there a reason for making the attributes protected instead of private with protected setters/getters?

Reply:
Will update it

Reply:
Fixed, also updates filteredItems, that was defined/used in the same way.

Reply:
Why do these methods have to be **public**? In 365a1cd67b080825c2fca63144751ff063911f70 `filter` and `filteredItemes` were protected. Why is `setFilter` needed? Can't `filter` be **final**?

```
  public TextField getFilter()
  public void setFilter(TextField filter)
  public List<T> getFilteredItems()
  public void setFilteredItems(List<T> filteredItems)

Reply:
Fixed. The setters are not needed and the getters don't need to be public. Also filter can be final.